be/src/exprs/string_functions.cpp

```diff
@@ -3236,4 +3238,58 @@ StatusOr<ColumnPtr> StringFunctions::parse_url(FunctionContext* context, const s
     return parse_url_general(context, columns);
 }
 
+StatusOr<ColumnPtr> StringFunctions::compress(FunctionContext* context, const starrocks::Columns& columns) {
+    auto str_viewer = ColumnViewer<TYPE_VARCHAR>(columns[0]);
+    auto size = columns[0]->size();
+    ColumnBuilder<TYPE_VARCHAR> result(size);
+    raw::RawString compression_scratch;
+    if (columns.size() == 1) {
+        const BlockCompressionCodec* compress_codec = nullptr;
+        RETURN_IF_ERROR(get_block_compression_codec(CompressionTypePB::ZLIB, &compress_codec));
+        for (int row = 0; row < size; ++row) {
+            if (str_viewer.is_null(row)) {
+                result.append_null();
+                continue;
+            }
+            const Slice str_slice = str_viewer.value(row);
+            int max_compressed_size = compress_codec->max_compressed_len(str_slice.get_size());
+            if (compression_scratch.size() < max_compressed_size) {
+                compression_scratch.resize(max_compressed_size);
+            }
+            Slice compressed_slice{compression_scratch.data(), compression_scratch.size()};
+            RETURN_IF_ERROR(compress_codec->compress(str_slice, &compressed_slice));
+            result.append(compressed_slice);
+        }
+    } else {
```

Comment:
For this case, the second part often be const values.
Would you like to optimize the common case, like writing a compress_prepare and compress_close to find the compressor once? Then the performance will be much better.

Reply:
> Would you like implement decompress function together?

OK, I will try to do it.